[Build] nvcc fatal : 'ON': expected a number

### Describe the issue

When I build onnxruntime with CUDA support, it tries to use:
```
nvcc ... --theads ON
```
which fails with:
```
nvcc fatal   : 'ON': expected a number
```

### Urgency

Not really, will build without CUDA support for now.

### Target platform

linux-ubuntu22.04-zen2

### Build script

```
$ spack install py-onnxruntime+cuda
```

### Error / output

[build log](https://github.com/microsoft/onnxruntime/files/14539108/spack-build-out.txt)


### Visual Studio Version

_No response_

### GCC / Compiler Version

12.3.0

Hey,

similar results here when trying to compile onnxruntime from source on a NVidia Jetson AGX Orin with Jetpack 6.0.

# Description
The build seems to fail halfway through the build on the cuda provider, giving the message `nvcc fatal   : 'ON': expected a number`
I am trying to compile onnxruntime 1.17.3 against cuda 12.2 (preinstalled version on the jetpack 6.0.0 release, as well as tensorrt).

For my application I need to be able to use tensorrt, due to the optimizations available.

# Urgency
To me: relatively high. :)
Our development would get stuck on this issue.

# Target system
`Linux ubuntu 5.15.136-tegra #1 SMP PREEMPT Mon May 6 09:56:39 PDT 2024 aarch64 aarch64 aarch64 GNU/Linux`
Jetpack 6.0.0

`# R36 (release), REVISION: 3.0, GCID: 36106755, BOARD: generic, EABI: aarch64, DATE: Thu Apr 25 03:14:05 UTC 2024
# KERNEL_VARIANT: oot`

# build script
```
cmake \
  -Donnxruntime_ENABLE_LANGUAGE_INTEROP_OPS=true \
  -Donnxruntime_ENABLE_CUDA_PROFILING=true \
  -Donnxruntime_USE_CUDA=true \
  -Donnxruntime_BUILD_FOR_NATIVE_MACHINE=true \
  -Donnxruntime_USE_TENSORRT=true \
  -Donnxruntime_BUILD_SHARED_LIB=true \
  -S /opt/onnxruntime/onnxruntime-1.17.3/cmake \
  -B /opt/onnxruntime/onnxruntime-1.17.3/build
```
# compiler version
nvcc --version
```
Cuda compilation tools, release 12.2, V12.2.140
Build cuda_12.2.r12.2/compiler.33191640_0
```

gcc --version
```
gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
```
# Error/output

 Terminal output:
```
[ 51%] Building CUDA object CMakeFiles/onnxruntime_providers_cuda.dir/opt/onnxruntime/onnxruntime-1.17.3/onnxruntime/core/providers/cuda/activation/activations_impl.cu.o
nvcc fatal   : 'ON': expected a number
gmake[2]: *** [CMakeFiles/onnxruntime_providers_cuda.dir/build.make:1365: CMakeFiles/onnxruntime_providers_cuda.dir/opt/onnxruntime/onnxruntime-1.17.3/onnxruntime/core/providers/cuda/activation/activations_impl.cu.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:2117: CMakeFiles/onnxruntime_providers_cuda.dir/all] Error 2
gmake: *** [Makefile:166: all] Error 2
```

makefile at the point it is pointing to:
```
/usr/local/cuda-12.2/bin/nvcc -forward-unknown-to-host-compiler $(CUDA_DEFINES) $(CUDA_INCLUDES) $(CUDA_FLAGS) -MD -MT CMakeFiles/onnxruntime_providers_cuda.dir/opt/onnxruntime/onnxruntime-1.17.3/onnxruntime/core/providers/cuda/activation/activations_impl.cu.o -MF CMakeFiles/onnxruntime_providers_cuda.dir/opt/onnxruntime/onnxruntime-1.17.3/onnxruntime/core/providers/cuda/activation/activations_impl.cu.o.d -x cu -c /opt/onnxruntime/onnxruntime-1.17.3/onnxruntime/core/providers/cuda/activation/activations_impl.cu -o CMakeFiles/onnxruntime_providers_cuda.dir/opt/onnxruntime/onnxruntime-1.17.3/onnxruntime/core/providers/cuda/activation/activations_impl.cu.o
```

--- a/cmake/onnxruntime_providers_cuda.cmake
+++ b/cmake/onnxruntime_providers_cuda.cmake
@@ -153,7 +153,7 @@
     # CUDA 11.3+ supports parallel compilation
     # https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#options-for-guiding-compiler-driver-threads
     if (CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 11.3)
-      option(onnxruntime_NVCC_THREADS "Number of threads that NVCC can use for compilation." 1)
+      set(onnxruntime_NVCC_THREADS "1" CACHE STRING "Number of threads that NVCC can use for compilation.")
       target_compile_options(${target} PRIVATE "$<$<COMPILE_LANGUAGE:CUDA>:SHELL:--threads \"${onnxruntime_NVCC_THREADS}\">")
     endif()
 

